ps/jenkins/percona-server-for-mysql-8.0.groovy

```diff
@@ -33,43 +33,55 @@ void installCli(String PLATFORM) {
-        if [ ${FIPSMODE} = "YES" ]; then
-            sed -i 's|percona-server-server/usr|percona-server-server-pro/usr|g' ps_builder.sh
-            sed -i 's|dbg-package=percona-server-dbg|dbg-package=percona-server-pro-dbg|g' ps_builder.sh
-        fi
-        grep "percona-server-server" ps_builder.sh
-        export build_dir=\$(pwd -P)
-        if [ "$DOCKER_OS" = "none" ]; then
-            set -o xtrace
-            cd \${build_dir}
-            if [ -f ./test/percona-server-8.0.properties ]; then
-                . ./test/percona-server-8.0.properties
-            fi
-            sudo bash -x ./ps_builder.sh --builddir=\${build_dir}/test --install_deps=1
-            if [ ${BUILD_TOKUDB_TOKUBACKUP} = "ON" ]; then
-                bash -x ./ps_builder.sh --builddir=\${build_dir}/test --repo=${GIT_REPO} --branch=${BRANCH} --build_tokudb_tokubackup=1 --perconaft_branch=${PERCONAFT_BRANCH} --tokubackup_branch=${TOKUBACKUP_BRANCH} --rpm_release=${RPM_RELEASE} --deb_release=${DEB_RELEASE} ${STAGE_PARAM}
-            else
-                bash -x ./ps_builder.sh --builddir=\${build_dir}/test --repo=${GIT_REPO} --branch=${BRANCH} --perconaft_branch=${PERCONAFT_BRANCH} --tokubackup_branch=${TOKUBACKUP_BRANCH} --rpm_release=${RPM_RELEASE} --deb_release=${DEB_RELEASE} ${STAGE_PARAM}
-            fi
-        else
-            docker run -u root -v \${build_dir}:\${build_dir} ${DOCKER_OS} sh -c "
-                set -o xtrace
-                cd \${build_dir}
-                if [ -f ./test/percona-server-8.0.properties ]; then
-                    . ./test/percona-server-8.0.properties
-                fi
-                bash -x ./ps_builder.sh --builddir=\${build_dir}/test --install_deps=1
-                if [ ${BUILD_TOKUDB_TOKUBACKUP} = \"ON\" ]; then
-                    bash -x ./ps_builder.sh --builddir=\${build_dir}/test --repo=${GIT_REPO} --branch=${BRANCH} --build_tokudb_tokubackup=1 --perconaft_branch=${PERCONAFT_BRANCH} --tokubackup_branch=${TOKUBACKUP_BRANCH} --rpm_release=${RPM_RELEASE} --deb_release=${DEB_RELEASE} ${STAGE_PARAM}
-                else
-                    bash -x ./ps_builder.sh --builddir=\${build_dir}/test --repo=${GIT_REPO} --branch=${BRANCH} --perconaft_branch=${PERCONAFT_BRANCH} --tokubackup_branch=${TOKUBACKUP_BRANCH} --rpm_release=${RPM_RELEASE} --deb_release=${DEB_RELEASE} ${STAGE_PARAM}
-                fi"
-        fi
-    """
+    withCredentials([string(credentialsId: 'GITHUB_API_TOKEN', variable: 'TOKEN')]) {
+      sh """
+          set -o xtrace
+          mkdir -p test
+          if [ \${FIPSMODE} = "YES" ]; then
+              PRO_BRANCH=\$(echo "${BRANCH}" | sed -E "s/release-([0-9]+\\.[0-9]+).*/\\1/")
+              curl -L -H "Authorization: Bearer \${TOKEN}" \
```

Comment:
Could it be verified if a PRO_BRANCH is provided and if it's not existed then use 8.0 or 8.4 branch.

![Screenshot 2025-02-12 at 13 16 39](https://github.com/user-attachments/assets/699626f2-a05d-4bb8-9b89-8126d9f778ce)


Please notice that there are no release-X.X.X branches.
![Screenshot 2025-02-12 at 13 20 57](https://github.com/user-attachments/assets/9388ddf4-98c0-4303-be60-eb468ccc7c59)